### ISSUE TITLE
[Bug] Manual input of derivation index does not accurately look up account address for account with rotated authentication key

### ISSUE THREAD

**Issue by alnoki** [aptos-labs/petra-wallet#47]
Related: #44 

When manually inputting the derivation index of a wallet, Petra appears not to check the `OriginatingAddress` table: this means that if the derivation index controls an account that has had its authentication key rotated, Petra will not allow the user to control their account. Instead it will try to use the account address that is generated from the derivation index's corresponding public key

**Comment by GhostWalker562** [comment]
Thanks for pointing this out. We are looking into this.

**Comment by GhostWalker562** [comment]
Fixed in `1.2.73`, on the import screen it may show the incorrect address but when importing it will derive the rotated address.

**Comment by alnoki** [comment]
@GhostWalker562 as of version `1.2.76` this problem is not resolved

Was this feature tested before closing the issue?

How can I force lookup of the rotated address?

**Comment by GhostWalker562** [comment]
@alnoki

The feature was tested as follows: 
1. Follow key rotation guide for Ledger: https://aptos.dev/en/build/cli/trying-things-on-chain/ledger#authentication-key-rotation
2. Import rotated Ledger account
3. Verify that the imported address targets to the correct account
4. Send coins

There is no "force lookup" option right now, you will need to reimport the account to target the correct address

Let me know if I am missing something, thanks!

**Comment by alnoki** [comment]
> @alnoki
> 
> The feature was tested as follows:
> 
> 1. Follow key rotation guide for Ledger: https://aptos.dev/en/build/cli/trying-things-on-chain/ledger#authentication-key-rotation
> 2. Import rotated Ledger account
> 3. Verify that the imported address targets to the correct account
> 4. Send coins
> 
> There is no "force lookup" option right now, you will need to reimport the account to target the correct address
> 
> Let me know if I am missing something, thanks!

@GhostWalker562  unfortunately this process does not work for the following workflow, which 

1. Create a new account with a random address, say `0xabc....`
2. Rotate `0xabc` to be secured by Ledger using derivation index 1000 via the CLI
3. Use Petra option to try to import Ledger for derivation index 1000 (per attached), which is an attempt to "reimport the account"

<img width="432" alt="Screenshot 2025-01-06 at 16 07 55" src="https://github.com/user-attachments/assets/155a4c02-561b-41a8-a917-7f1bd31d0312" />

In my experience, this does not import the `0xabc....` target account, but rather, it tries to import the account address that derivation index 1000 would correspond to *without* rotation. I believe this is because Petra is not checking the `OriginatingAddress` table

Are you able to update Petra to check the `OriginatingAddress` table? This is what the CLI does when importing Ledger accounts

**Comment by GhostWalker562** [comment]
@alnoki Made a mistake in my last reply. For "Step 2" I imported the Ledger account but meant to say that I rotated a hot wallet to be secured by that Ledger account as mentioned in the guides. Could you make sure that the wallet is on the correct network while importing? When we check for `OriginatingAddress`, it uses the current network the wallet is on. If the wallet is on a different network than the rotation, the `OriginatingAddress` table will not have the corresponding rotation.

**Comment by alnoki** [comment]
> @alnoki Made a mistake in my last reply. For "Step 2" I imported the Ledger account but meant to say that I rotated a hot wallet to be secured by that Ledger account as mentioned in the guides. Could you make sure that the wallet is on the correct network while importing? When we check for `OriginatingAddress`, it uses the current network the wallet is on. If the wallet is on a different network than the rotation, the `OriginatingAddress` table will not have the corresponding rotation.

@GhostWalker562 I have verified that they key rotation was performed on mainnet and that Petra had mainnet selected before trying to link a new Ledger account

**Comment by GhostWalker562** [comment]
Could you share your exact steps, including CLI commands to reproduce this issue? @alnoki 

Here are mine:
1. Generate private key
```
aptos key generate \
    --assume-yes \
    --output-file private-key-a \
    --vanity-prefix 0xaaa
```
2. Generate hot wallet, creates: `0xaaa6799167b3f856b0b51df3844af6503400a9f4fb435fd88170619d0cd223c7`
```
aptos init \
    --assume-yes \
    --network devnet \
    --private-key-file private-key-a \
    --profile hot-wallet-1
```
3. Rotate hot wallet to be secured by Ledger
```
aptos account rotate-key \
    --assume-yes \
    --new-derivation-index 1000 \
    --profile hot-wallet-1 \
    --save-to-profile ledger-wallet-1000
```
4. Approve on Ledger
5. Import Ledger to Petra with path `m/44'/637'/1000'/0'/0'`
<img width="461" alt="image" src="https://github.com/user-attachments/assets/0603cf29-b863-4034-91c3-b6b598967fb2" />

6. Confirms that address is `0xaaa6799167b3f856b0b51df3844af6503400a9f4fb435fd88170619d0cd223c7`
<img width="368" alt="image" src="https://github.com/user-attachments/assets/064b6b3b-c3b3-4456-a345-2e12f406c432" />

**Comment by alnoki** [comment]
@GhostWalker562 thank you for providing the steps. Note in step 5 that the address listed is `0xdd...bddc` instead of `0xaa..23c7`

Hence when I was performing this myself, I stopped at step 5 because I saw that the wrong address was listed, e.g. was not the results of the `OriginatingAddress` table: this is the faulty behavior I was referring to

Can you fix Petra so that in step 5 of your flow, the listed address instead shows `0xaa..23c7`, the address that is *actually* used when you finish importing?

**Comment by GhostWalker562** [comment]
@alnoki This issue should be fixed in `1.2.78`

**Comment by alnoki** [comment]
> [@alnoki](https://github.com/alnoki) This issue should be fixed in `1.2.78`

@GhostWalker562 The rotated address is now correct, but the balance is not - is the balance looked up using the rotated one (`0xaa..23c7` in your example) or the other one (`0xdd...bddc` in your example)?

**Comment by GhostWalker562** [comment]
@alnoki Thanks for sharing this, and you are correct here, pushed out a fix and will be released with `1.2.81`
